### PR TITLE
Add test using metric class and metric instance

### PR DESCRIPTION
**Pull request by Flamefire**
Also contains some changes to the `make_exception` functions as I'm using them for the tests.

### CODE REVIEW

**Comment by bmario** [comment]
Please keep different concerns separated in different PR's. This PR consists of 3 completely unrelated commits.

**Comment by Flamefire** [comment]
Split into #20 and #21. Will rebase this when the others are merged.

File: tests/metric_events.cpp [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    def::metric_instance mInstance(0, mClass, location, location);
+    def::metric_member mMember(0, strings[4], strings[5], common::metric_type::other,
+                               common::metric_mode::accumulated_point, common::type::uint8,
+                               common::base_type::binary, 0, strings[6]);
+    mClass.add_member(mMember);
+    TEST(mInstance.metric_class() == mClass);
+
+    otf2::event::metric ev(otf2::chrono::genesis(), mInstance);
+
+    TEST(!ev.has_metric_class());
```
Comment (bmario) [inline tests/metric_events.cpp:69]: If you want that, please just add Catch2

Comment (Flamefire) [reply tests/metric_events.cpp:69]: I did not want to add another dependency and keep this as small as possible. Catch2 is great, but for now IMO we can live with a simple check-and-throw implemented on 5 lines.

**Pull request closed without merging**